lib/modules/manager/puppet/extract.ts

```diff
@@ -0,0 +1,115 @@
+import { logger } from '../../../logger';
+import { parseUrl } from '../../../util/url';
+import { GitTagsDatasource } from '../../datasource/git-tags';
+import { GithubTagsDatasource } from '../../datasource/github-tags';
+import { PuppetForgeDatasource } from '../../datasource/puppet-forge';
+import type { PackageDependency, PackageFile } from '../types';
+import { isGithubUrl, parseGitOwnerRepo } from './common';
+import { parsePuppetfile } from './puppetfile-parser';
+import type { PuppetfileModule } from './types';
+
+function parseForgeDependency(
+  module: PuppetfileModule,
+  forgeUrl?: string
+): PackageDependency {
+  const dep: PackageDependency = {
+    depName: module.name,
+    datasource: PuppetForgeDatasource.id,
+    packageName: module.name,
+    currentValue: module.version,
+  };
+
+  if (forgeUrl) {
+    dep.registryUrls = [forgeUrl];
+  }
+
+  return dep;
+}
+
+function parseGitDependency(module: PuppetfileModule): PackageDependency {
+  const moduleName = module.name;
+
+  const git = module.tags?.get('git');
+  const tag = module.tags?.get('tag');
+
+  if (!git || !tag) {
+    return {
+      depName: moduleName,
+      gitRef: true,
+      sourceUrl: git,
+      skipReason: 'invalid-version',
+    };
+  }
+
+  const parsedUrl = parseUrl(git);
+  const githubUrl = isGithubUrl(git, parsedUrl || undefined);
+
+  if (githubUrl && parsedUrl && parsedUrl.protocol !== 'https:') {
```

Comment:
```suggestion
  if (githubUrl && parsedUrl?.protocol !== 'https:') {
```

Reply:
is not the same
| nr | git | githubUrl  | parsedUrl  | parsedUrl?.protocol  | githubUrl && parsedUrl && parsedUrl.protocol !== 'https:'  | githubUrl && parsedUrl?.protocol !== 'https:'  |
|---|---|---|---|---|---|---|
| 1 | https://github.com.example.com/puppetlabs/puppetlabs-apache  |  false |  https://github.com.example.com/puppetlabs/puppetlabs-apache  |  https  |  false |  false | 
| 2 | http://github.com.example.com/puppetlabs/puppetlabs-apache  |  true  | https://github.com.example.com/puppetlabs/puppetlabs-apache  |  http  |  true  |  true  | 
| 3 | git@github.com:puppetlabs/puppetlabs-stdlib.git  |  true  | null  |  undefined  |  false  |  true  | 

in case 3 we get different results, because the ssh-url is not parsale (is null) and ```parseUrl?.protocol``` is not equal to ```parseUrl && parseUrl.protocol```